src/core/Events.js

```diff
@@ -115,6 +115,13 @@ L.Evented = L.Class.extend({
 			// so simple array makes the memory footprint better while not degrading performance
 
 			events[type] = events[type] || [];
+			for (var i = 0, len = events[type].length; i < len; i++) {

```

Comment:
While this seems like the only way to do it without some larger refactor of events, I'm not to fond of this change.

As I understand it, we handle the events code like "performance critical", it contains some optimizations to be fast.

With this change, adding _n_ listeners to an object changes from an _O(n)_ operation to an _O(n^2)_ operation, which is something I assume we want to avoid.

The other option would be to handle contextless and `this`-scoped listeners the same way as listeners with context, but that would also have performance implications.

Do we have any tests or other means to evaluate the performance differences in practice?